jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java

```diff
@@ -18,13 +18,13 @@ public AbstractJGivenReportImpl( String name, Task task, String relativeEntryPat
     }
 
     @Override public void configure( ReportGenerator generator ) {
-        generator.setTargetDirectory( getDestination() );
-        generator.setFormat( getFormat() );
-        generator.getConfig().setCustomCssFile( getCustomCssFile() );
-        generator.getConfig().setCustomJsFile( getCustomJsFile() );
-        generator.getConfig().setExcludeEmptyScenarios( isExcludeEmptyScenarios() );
-        if (getTitle() != null) {
-            generator.getConfig().setTitle( getTitle() );
+        generator.addFlag( "--targetDir=" + getDestination() );
```

Comment:
The problem I see is that you have more or less introduced the command-line frontend into the backend code. This is typically a bad idea. It would be nicer to have clean separation of the frontend (parsing) and the backend. So that the backend does not have to know anything about how command line arguments are parsed/have to look like etc. So my suggestion would be the following:
- Introduce a class for a command line option, something like:

      class CommandLineOption {
           String longName; // mandatory, can be used as key later
           String shortName;
           String description;
           boolean hasValue;
           boolean optional;
           Object defaultValue;
      }

- Have a generic method in the AbstractReportGenerator to specify additional CommandLineOption 
- Implement a CommandLineOptionParser that takes a list of CommandLineOption and an array of Strings and returns a map from String to Object, where String is the longName of the argument and Object the parsed value.

- Have a generic method (in the command line parser) to get the value for an CommandLineOption like:
  
      public Object getValue( CommandLineOption option) {
            return parsedOptions.get(option.getLongName());
      }

If we are at it already, we could even go a step further and generalize this to completely abstract away where the configuration options come from. Something similar to what the TNG config-builder does: https://github.com/TNG/config-builder (but simpler, not using annotations etc). Because in the end, we also want to provide a way to configure the cssFile, for example, using properties. So we may need a general configuration option class like:

      class ConfigurationOption {
             String longName;
             String shortName;
             String propertyString;
             String envString;
             boolean optional;
             boolean hasValue;
             String defaultValue;
             ...
      }

You should also provide a Builder class for that so that it easier to create instances of that class. And instead of only getting these options by using the command line arguments, we would also look for properties and environment variables.

In the end however, there should also be a class like `ReportConfig` and `HtmlReportConfig` that only contain fields with the final configuration values:

      class ReportConfig {
             File targetDirectory;
             ...
      }

      class HtmlReportConfig extends ReportConfig {
             File cssFile;
      }

This is can then be used by the backend code.